### PR TITLE
jsc: TypedArray indexed access at 4294967295 on a 2**32-length view

### PR DESCRIPTION
**Pull request by robobun**
## Problem

```js
const u = new Uint8Array(2**32);
u.length;                        // 4294967296
u[4294967294];                   // 0
u[4294967295];                   // undefined   (V8: 0)
u[4294967295] = 42;
u[4294967295];                   // undefined   (V8: 42)
u.at(4294967295);                // undefined   (V8: 42)
4294967295 in u;                 // true        ([[HasProperty]] disagrees with [[Get]])
Object.hasOwn(u, "4294967295");  // false       (V8: true)
Object.getOwnPropertyDescriptor(u, 4294967295);  // undefined
Object.defineProperty(u, 4294967295, { value: 1, writable: true, enumerable: true, configurable: true });
  // TypeError: Attempting to store canonical numeric string property on a typed array
new DataView(u.buffer).getUint8(4294967295);     // 0 (DataView path is fine)
```

[`IsValidIntegerIndex(O, index)`](https://tc39.es/ecma262/#sec-isvalidintegerindex) only requires `0 <= index < O.[[ArrayLength]]`. A `Uint8Array` can have length up to `MAX_ARRAY_BUFFER_SIZE` (`2**32` on 64-bit), so `4294967295` is a valid index there, and V8 treats it as one.

## Cause

`JSGenericTypedArrayView::{getOwnPropertySlot, put, defineOwnProperty, deleteProperty}(PropertyName)` call `parseIndex()`, which is capped at `MAX_ARRAY_INDEX = 0xFFFFFFFE` because a regular JS Array's length is a uint32 and the largest index is `length - 1`. For `"4294967295"` `parseIndex` returns `nullopt`, and the `isCanonicalNumericIndexString` fallback treats it as always out of bounds. The `unsigned`-index overloads (`getOwnPropertySlotByIndex` etc.) already compare against the full `size_t` length via `inBounds()`, which is why `4294967295 in u` (routed through `getUInt32` → `hasProperty(unsigned)`) returns `true` while `u[4294967295]` returns `undefined`.

## Fix

[oven-sh/WebKit#348](https://github.com/oven-sh/WebKit/pull/348) adds `parseTypedArrayIndex(PropertyName)` in `PropertyName.h` that extends `parseIndex()` to the full uint32 range (`UINT32_MAX` is the only extra value, handled as a literal compare after `parseIndex` returns `nullopt`) and routes the four `JSGenericTypedArrayView` PropertyName overloads through it.

TypedArray structures are shared per type, not per length, so once `getOwnPropertySlot("4294967295")` becomes length-dependent the structure-keyed caches that previously memoised "always absent" must be told not to: `HasOwnPropertyCache::tryAdd` widens its index guard to `parseTypedArrayIndex`, and the `isTypedArrayType && isCanonicalNumericIndexString` short-circuits in `generateConditions` / `prepareChainForCaching` / `PolyProtoAccessChain::tryCreate` return invalid instead of a valid empty condition set when `parseTypedArrayIndex` accepts the key, so no `Miss`/`InMiss` IC is installed for it on a TypedArray. Without those guards, `Object.hasOwn(short, "4294967295"); Object.hasOwn(long, "4294967295")` trips `ASSERT(*result == hasOwnProperty(...))` in `ObjectPrototype.cpp` on assert-enabled builds, and a warmed IC returns the stale miss in release builds.

The DFG/FTL inline fast paths speculate Int32 for the index and OSR-exit or fall to the C++ slow path for `0xFFFFFFFF`, so no JIT change is needed.

This PR bumps `WEBKIT_VERSION` to the `autobuild-preview-pr-348-c106f67d` preview build and adds `test/js/bun/jsc/typedarray-uint32max-index.test.ts`, which spawns a child that allocates a `2**32`-byte `Uint8Array` and exercises `[[Get]]`, `[[Set]]`, `.at`, `in`, `Object.hasOwn`, `Object.getOwnPropertyDescriptor`, `Object.defineProperty`, `Reflect.set` and `Reflect.deleteProperty` at index `4294967295`, the same key on a short view (still out of bounds), the adjacent canonical numeric strings `"4294967296"` / `"-0"` (still `undefined`), `Int8Array`/`Uint8ClampedArray` views over the same buffer, and the `hasOwn(small)→hasOwn(big)` / warmed-IC `probe(small)→probe(big)` orderings that exercise the cache guards. The child skips cleanly if the allocation is refused.

## Also in this range (549170099226..c106f67d)

- [oven-sh/WebKit#328](https://github.com/oven-sh/WebKit/pull/328) inspector: release throw scope before tail-calling impl in injected-script prototype host functions
- [oven-sh/WebKit#317](https://github.com/oven-sh/WebKit/pull/317) LiteralParser: throw RangeError on OOM when copying a JSON string value
- [oven-sh/WebKit#331](https://github.com/oven-sh/WebKit/pull/331) SignalsWin: fix VEH return value and register behind AddressSanitizer's handler
- [oven-sh/WebKit#332](https://github.com/oven-sh/WebKit/pull/332) Heap: make `minEdenToOldGenerationRatio` a JSC option (same default; behaviour unchanged)
- [oven-sh/WebKit#347](https://github.com/oven-sh/WebKit/pull/347) getAsyncStackTrace: unwrap InternalFieldTuple in reaction context

## Verification

```
$ USE_SYSTEM_BUN=1 bun test test/js/bun/jsc/typedarray-uint32max-index.test.ts
(fail)    # initialGetNum undefined, afterDefine "threw: TypeError", hasOwn false, ...

$ bun bd test test/js/bun/jsc/typedarray-uint32max-index.test.ts
(pass)
```

Node 26.3.0 on the same fixture prints the direct-access values exactly. For the warmed-IC ordering Node returns `undefined`/`false` (V8 shares the stale-miss behaviour), which is a V8 bug against `IsValidIntegerIndex`; this change gives the spec answer there.

Found while fixing `Buffer.writeUInt8` rejecting offset `4294967295` (#35867 works around it at the Buffer layer; this closes the engine-level gap).

Once oven-sh/WebKit#348 merges, `WEBKIT_VERSION` should be repointed at the merged main sha (the preview release is deleted at that point).

---

**[decide:webkit]** gate passed · iteration 0 · 2 files touched

<details><summary>passes on PR (with fix)</summary>

```console
Test-only change.

Debug/ASAN (expected pass):
$ bun bd test 'test/js/bun/jsc/typedarray-uint32max-index.test.ts'
$ BUN_DEBUG_QUIET_LOGS=1 bun scripts/build.ts --profile=debug --quiet test "test/js/bun/jsc/typedarray-uint32max-index.test.ts"
bun test v1.4.0 (e83092d93)

test/js/bun/jsc/typedarray-uint32max-index.test.ts:
(pass) TypedArray indexed access at 4294967295 on a 2**32-length view [420.38ms]

 1 pass
 0 fail
 3 expect() calls
Ran 1 test across 1 file. [2.79s]
Exit: 0
```

</details>

<details><summary>diff hotspot</summary>

```
scripts/build/deps/webkit.ts                       |   2 +-
 test/js/bun/jsc/typedarray-uint32max-index.test.ts | 146 +++++++++++++++++++++
 2 files changed, 147 insertions(+), 1 deletion(-)
```

</details>

**gate history** · 1 passed · 0 rejected · iteration 0

<details><summary>evidence per changed file</summary>

```
file                                                reads  edits  tests
scripts/build/deps/webkit.ts                            1      2      0
test/js/bun/jsc/typedarray-uint32max-index.test.ts      2      6      0
```

</details>

### CODE REVIEW

File: test/js/bun/jsc/typedarray-uint32max-index.test.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+console.log(JSON.stringify(out));
+`;
+
+test("TypedArray indexed access at 4294967295 on a 2**32-length view", async () => {
+  await using proc = Bun.spawn({
+    cmd: [bunExe(), "-e", fixture],
+    env: bunEnv,
+    stdout: "pipe",
+    stderr: "pipe",
```
[the remark this answers is not in the document]
Comment (robobun) [reply test/js/bun/jsc/typedarray-uint32max-index.test.ts:106]: Applied in a901bb309a, matching the pattern from md-edge-cases/arraybuffersink/env/fs-oom tests.
